drivers/npmx/npmx_shell.c

```diff
@@ -2384,1723 +2474,1185 @@ static npmx_ldsw_gpio_t ldsw_gpio_index_convert(int8_t gpio_idx)
-
-	if (err != 0) {
-		shell_error(shell, "Error: GPIO number has to be an integer.");
+	npmx_ldsw_t *ldsw_instance = ldsw_instance_get(shell, args_info.arg[0].result.uvalue);
+	if (ldsw_instance == NULL) {
 		return 0;
 	}
 
-	if (gpio_idx >= NPM_GPIOS_COUNT) {
-		shell_error(shell, "Error: GPIO number is too high: no such instance.");
+	npmx_ldsw_soft_start_config_t soft_start_config;
+	npmx_error_t err_code = npmx_ldsw_soft_start_config_get(ldsw_instance, &soft_start_config);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "soft-start config");
 		return 0;
 	}
 
-	npmx_gpio_t *gpio_instance = npmx_gpio_get(npmx_instance, gpio_idx);
-
-	bool gpio_status;
-	npmx_error_t err_code = npmx_gpio_status_check(gpio_instance, &gpio_status);
-
-	if (!check_error_code(shell, err_code)) {
-		shell_error(shell, "Error: unable to read GPIO status.");
-	} else {
-		shell_print(shell, "Value: %u.", (uint8_t)gpio_status);
+	switch (config_type) {
+	case LDSW_SOFT_START_PARAM_ENABLE:
+		print_value(shell, soft_start_config.enable, UNIT_TYPE_NONE);
+		break;
+	case LDSW_SOFT_START_PARAM_CURRENT:
+		uint32_t soft_start_current_ma;
+		if (!npmx_ldsw_soft_start_current_convert_to_ma(soft_start_config.current,
+								&soft_start_current_ma)) {
+			print_convert_error(shell, "soft-start current", "milliamperes");
+			return 0;
+		}
+		print_value(shell, soft_start_current_ma, UNIT_TYPE_MILLIAMPERE);
+		break;
 	}
-
 	return 0;
 }
 
-static npmx_gpio_mode_t gpio_mode_convert(uint8_t mode)
+static int cmd_ldsw_soft_start_current_set(const struct shell *shell, size_t argc, char **argv)
 {
-	switch (mode) {
-	case 0:
-		return NPMX_GPIO_MODE_INPUT;
-	case 1:
-		return NPMX_GPIO_MODE_INPUT_OVERRIDE_1;
-	case 2:
-		return NPMX_GPIO_MODE_INPUT_OVERRIDE_0;
-	case 3:
-		return NPMX_GPIO_MODE_INPUT_RISING_EDGE;
-	case 4:
-		return NPMX_GPIO_MODE_INPUT_FALLING_EDGE;
-	case 5:
-		return NPMX_GPIO_MODE_OUTPUT_IRQ;
-	case 6:
-		return NPMX_GPIO_MODE_OUTPUT_RESET;
-	case 7:
-		return NPMX_GPIO_MODE_OUTPUT_PLW;
-	case 8:
-		return NPMX_GPIO_MODE_OUTPUT_OVERRIDE_1;
-	case 9:
-		return NPMX_GPIO_MODE_OUTPUT_OVERRIDE_0;
-	default:
-		return NPMX_GPIO_MODE_INVALID;
-	}
+	return ldsw_soft_start_config_set(shell, argc, argv, LDSW_SOFT_START_PARAM_CURRENT);
 }
 
-static npmx_gpio_pull_t gpio_pull_convert(uint8_t pull)
+static int cmd_ldsw_soft_start_current_get(const struct shell *shell, size_t argc, char **argv)
 {
-	switch (pull) {
-	case 0:
-		return NPMX_GPIO_PULL_DOWN;
-	case 1:
-		return NPMX_GPIO_PULL_UP;
-	case 2:
-		return NPMX_GPIO_PULL_NONE;
-	default:
-		return NPMX_GPIO_PULL_INVALID;
-	}
+	return ldsw_soft_start_config_get(shell, argc, argv, LDSW_SOFT_START_PARAM_CURRENT);
 }
 
-static npmx_error_t gpio_config_type_set(const struct shell *shell, gpio_config_type_t config_type,
-					 npmx_gpio_config_t *gpio_config, uint32_t input_arg)
+static int cmd_ldsw_soft_start_status_set(const struct shell *shell, size_t argc, char **argv)
```

Comment:
I don't like
npmx ldsw soft_start enable -> npmx ldsw soft_start status
This enables/disables soft_start. It doesn't return any status.
enable is left in the code
case LDSW_SOFT_START_PARAM_**ENABLE**:
    soft_start_config.**enable** = result.bvalue;
    break;